tests/rptest/services/redpanda.py

```diff
@@ -2891,14 +2891,18 @@ def remove_local_data(self, node):
 
     def redpanda_pid(self, node):
         try:
-            cmd = "pgrep --exact redpanda"
-
-            for p in node.account.ssh_capture(cmd,
-                                              allow_fail=True,
-                                              callback=int,
-                                              timeout_sec=10):
-                return p
-
+            cmd = "pgrep --list-full --exact redpanda"
```

Comment:
Another thought I had is to use -f mode and look for the exact pattern we intentionally inject in the start command..

say we include an env variable like REDPANDA_PID=1...  


```
REDPANDA_PID=1 ASAN_OPTIONS=abort_on_error=1:disable_coredump=0:unmap_shadow_on_exit=1 nohup /work/redpanda/vbuild/debug/clang/dist/local/redpanda/bin/redpanda --redpanda-cfg /etc/redpa       nda/redpanda.yaml --default-log-level trace --logger-log-level=exception=debug:io=debug:seastar_memory=debug  --abort-on-seastar-bad-alloc  --dump-memory-diagnostics-on-alloc-failure-kind=all  --kernel-page-cache=true --overprovisioned  --reserve-memory=0M --smp=2 --memory=2048M --unsafe-bypass-fsync=0         >> /var/lib/redpanda/redpanda.log 2>&1 &
```

and look for something like

pgrep -f "REDPANDA_PID.\*redpanda.\*--redpanda-cfg.*"